homeassistant/components/unifi/entity.py

```diff
@@ -0,0 +1,176 @@
+
+    allowed_fn: Callable[[UniFiController, str], bool]
+    api_handler_fn: Callable[[aiounifi.Controller], HandlerT]
+    available_fn: Callable[[UniFiController, str], bool]
+    device_info_fn: Callable[[aiounifi.Controller, str], DeviceInfo | None]
+    event_is_on: tuple[EventKey, ...] | None
+    event_to_subscribe: tuple[EventKey, ...] | None
+    name_fn: Callable[[DataT], str | None]
+    object_fn: Callable[[aiounifi.Controller, str], DataT]
+    supported_fn: Callable[[UniFiController, str], bool | None]
+    unique_id_fn: Callable[[UniFiController, str], str]
+
+
+@dataclass
+class UnifiEntityDescription(EntityDescription, UnifiDescription[HandlerT, DataT]):
+    """UniFi Entity Description."""
+
+
+class UnifiEntity(Entity, Generic[HandlerT, DataT]):
+    """Representation of a UniFi entity."""
+
+    entity_description: UnifiEntityDescription[HandlerT, DataT]
+    _attr_should_poll = False
+
+    _attr_unique_id: str
+
+    def __init__(
+        self,
+        obj_id: str,
+        controller: UniFiController,
+        description: UnifiEntityDescription[HandlerT, DataT],
+    ) -> None:
+        """Set up UniFi switch entity."""
+        self._obj_id = obj_id
+        self.controller = controller
+        self.entity_description = description
+
+        self._removed = False
+
+        self._attr_available = description.available_fn(controller, obj_id)
+        self._attr_device_info = description.device_info_fn(controller.api, obj_id)
+        self._attr_unique_id = description.unique_id_fn(controller, obj_id)
+
+        obj = description.object_fn(self.controller.api, obj_id)
+        self._attr_name = description.name_fn(obj)
+        self.async_initiate_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+        description = self.entity_description
+        handler = description.api_handler_fn(self.controller.api)
+
+        # New data from handler
+        self.async_on_remove(
+            handler.subscribe(
+                self.async_signalling_callback,
+            )
+        )
+
+        # State change from controller or websocket
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                self.controller.signal_reachable,
+                self.async_signal_reachable_callback,
+            )
+        )
+
+        # Config entry options updated
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                self.controller.signal_options_update,
+                self.async_signal_options_updated,
+            )
+        )
+
+        # Subscribe to events if defined
+        if description.event_to_subscribe is not None:
+            self.async_on_remove(
+                self.controller.api.events.subscribe(
+                    self.async_event_callback,
+                    description.event_to_subscribe,
+                )
+            )
```

Comment:
Is this dead code, which is intended to be used in the future by an entity which has `event_to_subscribe`?

Reply:
Yes exactly, it is used by other platforms in my bigger branch which is too big to publish :)

Reply:
Maybe leave out the dead code for now and add it when it will be used?

Reply:
Sure. Next PR as soon as this has been approved ;)

Reply:
Done